src/webapi/event.rs

```diff
@@ -121,6 +315,212 @@ reference_boilerplate! {
 ///
 /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent)
 pub trait IMouseEvent: IUiEvent {
+    /// Returns whether the Alt key was down when this event was fired.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/altKey)
+    #[inline]
+    fn alt_key( &self ) -> bool {
+        js!(
+            return @{self.as_ref()}.altKey;
+        ).try_into().unwrap()
+    }
+
+    /// Indicates the mouse button that fired this event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button)
+    fn button( &self ) -> MouseButton {
+        match js!(
+            return @{self.as_ref()},.button;
+        ).try_into().unwrap() {
+            0 => MouseButton::Left,
+            1 => MouseButton::Wheel,
+            2 => MouseButton::Right,
+            3 => MouseButton::Button4,
+            4 => MouseButton::Button5,
+            _ => unreachable!("Unexpected MouseEvent.button value"),
+        }
+    }
+
+    /// Indicates which mouse buttons were down when this event was fired.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons)
+    fn buttons( &self ) -> Vec<MouseButton> {
```

Comment:
Hmm... I'm not sure how I feel about making this return an vector. The original JS API returns a bitmask, and it would be nice to transfer this to Rust, but in a type safe matter.

How about creating a `MouseButtons` type which we would return from this function instead of a raw vector? Then we wouldn't force ourselves to always return a vector here, and we could still expose this functionality as `MouseButtons::to_vec` while we could postpone having to figure out how to best deal with those flags in an idiomatic manner.